src/test/java/htsjdk/samtools/seekablestream/SeekableBufferedStreamTest.java

```diff
@@ -123,6 +124,43 @@ public void testSkip() throws IOException {
         }
     }
 
+    @Test
+    public void testSeek() throws IOException {
+        int bufferSize = 20000;
+        int startPosition = 250000;
+        int length = 5000;
+
+        final int[] RELATIVE_SEEK_OFFSET = new int[]{-bufferSize*2, -bufferSize, -bufferSize/2, -length, -length/2, 0,
+                length/2, length, bufferSize/2, bufferSize, bufferSize*2};
+
+        for (final int seekOffset : RELATIVE_SEEK_OFFSET) {
+            byte[] buffer1 = new byte[length];
+            SeekableStream unBufferedStream = new SeekableFileStream(BAM_FILE);
+            unBufferedStream.seek(startPosition);
+            int bytesRead = unBufferedStream.read(buffer1, 0, length);
+            Assert.assertEquals(length, bytesRead);
```

Comment:
It would be nice if we were explicitly checking that we were doing the cheap jumps when possible instead of expensive seeks.  It seems like it could be done by instrumenting the SeekableStream before we wrap it. Maybe not worth it though.

Reply:
This turned out to be easy to implement by checking if the internal buffer's object identity changed.